dns/dns_updater.go

```diff
@@ -146,7 +148,11 @@ func (u *updater) Update(update controller.IngressUpdate) error {
 		failedCount.Inc()
 		return err
 	}
-	recordsGauge.Set(float64(len(aRecords)))
+
+	if u.onlyDelAssoc {
+		aRecords = u.filterUnassociated(aRecords)
+		recordsGauge.Set(float64(len(aRecords)))
+	}
```

Comment:
That's a gauge that gets shipped off to Prometheus and I don't think any of the Prometheus stuff is being tested.